projects/fluid_simulation/shaders.h

```diff
@@ -58,21 +57,29 @@ class Texture
 {
 public:
     // Initialize a new empty texture.
-    // sim      Pointer to the main simulator instance (for accessing global state).
     // name     Name of the texture.
     // width    Texture width.
     // height   Texture height.
     // format   Texture format.
-    Texture(FluidSimulation* sim, const std::string& name, uint32_t width, uint32_t height, ppx::grfx::Format format);
+    // device   Device to use to create the storage and sampled views.
+    Texture(const std::string& name, uint32_t width, uint32_t height, ppx::grfx::Format format, ppx::grfx::DevicePtr device);
 
     // Initialize a new texture from an image file.
-    // sim      Pointer to the main simulator instance (for accessing global state).
     // fileName Image file to load.
-    Texture(FluidSimulation* sim, const std::string& fileName);
+    // device   Device to use to create the storage and sampled views.
+    Texture(const std::string& fileName, ppx::grfx::DevicePtr device);
+
+    uint32_t GetWidth() const { return mTexture->GetWidth(); }
+    uint32_t GetHeight() const { return mTexture->GetHeight(); }
+
+    // Compute and return the size of the texture normalized to the resolution.
+    // given in pixels.  This maps the size of the texture to the normalized
+    // coordinates ([-1, 1], [-1, 1]).
+    ppx::float2 GetNormalizedSize(ppx::uint2 resolution) const
+    {
+        return ppx::float2(GetWidth() * 2.0f / static_cast<float>(resolution.x), GetHeight() * 2.0f / static_cast<float>(resolution.y));
```

Comment:
Not sure if I follow the math here, are the values supposed to be in the range of [-1, 1]? 
If the width is equal to resolution wouldn't it be 2?

Reply:
Yes.  Resolution is 2, since it spans from -1 to +1.

Reply:
Got confused too, definitely thought a -1 was missing since the comments mentions the interval `[-1;1]`.
Maybe the comment should be rephrased?

Reply:
Sure. I'll update the comment.  Thanks.